### DISCUSSION TITLE
How to trigger a backup when new packages are published?

### DISCUSSION THREAD

**Discussion by ms2d** [verdaccio/verdaccio#3596] [Q&A]
I want to trigger a backup script each time a new packages is successfully published to Verdaccio, which would maintain a rolling backup of my private npm packages. Would this be a good use for the Notification feature? Or is there another recommendation or best practice?

Thanks!

**Comment by juanpicado** [comment]
There is a hook functionality that triggers every time a package mis published, it works as a endpoint execution but you could have a little server that every time an endpoint is triggered trigger a backup. Check docs for how it works.

**Comment by ms2d** [comment]
Ah perfect.

I had not thought of that, but of course! A simple Express.js server with a route for the POST request from Verdaccio should work. [Here is the link](https://verdaccio.org/docs/notifications) to the docs for the `notify` hook.

Thanks @juanpicado!
